platform/solr/solr-factory/src/main/java/org/codice/solr/client/solrj/SolrClient.java

```diff
@@ -0,0 +1,1030 @@
+   *
+   * @param timeout the maximum time to wait
+   * @param unit the time unit of the timeout argument
+   * @return <code>false</code> if the Solr server this client is connected or the corresponding
+   *     core is not available; <code>true</code> otherwise
+   * @throws InterruptedException if the current thread was interrupted while waiting
+   */
+  public boolean isAvailable(long timeout, TimeUnit unit) throws InterruptedException;
+
+  /**
+   * This method is the same as {@link #isAvailable()} but allows a caller to provide a listener
+   * that will be called back with a boolean flag representing the new availability for this Solr
+   * client whenever it changes in the future. The listener will also be called back right away with
+   * the current availability of the client. This method is different from an active ping as it
+   * might be basing its initial answer on cached information. It might, however, initiate a
+   * background ping.
+   *
+   * <p><i>Note:</i> Nothing will happen and the method will return right away if the client was
+   * already closed.
+   *
+   * <p><i>Note:</i> Returning <code>true</code> doesn't guarantee the next call to any methods will
+   * succeed as the server or the core might become unavailable at any time without notice.
+   *
+   * @param listener a listener to notify whenever the Solr client's availability changes
+   * @return <code>false</code> if the Solr server this client is connected or the corresponding
+   *     core is not available; <code>true</code> otherwise
+   */
+  public boolean isAvailable(Listener listener);
+
+  /**
+   * Registers an initializer to be called when this Solr client finishes initializing and becomes
+   * available. The initializer is called right away if this client is already available. The
+   * initializer will never be called twice.
+   *
+   * <p><i>Note:</i> Nothing will happen and the method will return right away if the client was
+   * already closed.
+   *
+   * <p><i>Note:</i> The client might no longer be available by the time the initializer is called
+   * as the server or the core might become unavailable at any time without notice. It is, however,
+   * guaranteed that the client was available at least once.
+   *
+   * @param initializer an initializer to notify when the Solr client first become available
+   */
+  public void whenAvailable(Initializer initializer);
+
+  /**
+   * Adds a collection of documents
+   *
+   * @param collection the Solr collection to add documents to
+   * @param docs the collection of documents
+   * @return an {@link UpdateResponse} from the server
+   * @throws IOException if there is a communication error with the server
+   * @throws SolrServerException if there is an error on the server
+   * @throws UnavailableSolrException if the Solr server or the core is unavailable
+   */
+  public UpdateResponse add(String collection, Collection<SolrInputDocument> docs)
```

Comment:
🤔 ⭕️

Reply:
Since this interface will most likely change and be reduced in functionality, I think it will evolve to the point where it will become different than the SolrJ one. As such, I would rather not add more to what will have to be changed later.

Reply:
Since this interface will mot likely evolve, I would rather keep it simple for now.